src/dipdup/performance.py

```diff
@@ -181,137 +180,152 @@ def stats(self) -> dict[str, Any]:
         return stats
 
 
-@dataclass
+@dataclass(config=ConfigDict(arbitrary_types_allowed=True))
 class _MetricManager:
-    # Copied from prometheus.py, probably should be removed
-    enabled: bool = False
+    # NOTE: Some metrics types are unions with int and float to make mypy happy
 
     # NOTE: General metrics
-    # NOTE: Transformed to prometheus metrics easily
-    levels_indexed = Counter('dipdup_levels_indexed_total', 'Total number of levels indexed')
-    levels_nonempty = Counter('dipdup_levels_nonempty_total', 'Total number of nonempty levels indexed')
-    levels_total = Counter('dipdup_levels_total', 'Total number of levels')
-    objects_indexed = Counter('dipdup_objects_indexed_total', 'Total number of objects indexed')
+    levels_indexed: Gauge | int = Gauge('dipdup_levels_indexed_total', 'Total number of levels indexed')
+    levels_nonempty: Counter = Counter('dipdup_levels_nonempty_total', 'Total number of nonempty levels indexed')
+    levels_total: Gauge | int = Gauge('dipdup_levels_total', 'Total number of levels')
+    objects_indexed: Counter = Counter('dipdup_objects_indexed_total', 'Total number of objects indexed')
+
+    # NOTE: Index metrics
+    handlers_matched: Gauge = Gauge('dipdup_index_handlers_matched_total', 'Index total hits', ['handler'])
```

Comment:
Should be a counter, since it's a cumulative count of matches.

Reply:
Noted